lib/matplotlib/patches.py

```diff
@@ -1906,6 +1905,13 @@ def register(klass, name, style):
         klass._style_list[name] = style
 
 
+def _register_style(style_list, cls=None, *, name=None):
```

Comment:
In my view it's more or less the same reason why
```
@property
def foo(self): ...
```
is better than
```
def foo(self): ...
foo = property(foo)
```
i.e. you immediately declare one of the most important things you're going to do with the newly declared function/class (in our cases, stash it in the style dict).

Added a docstring.

Reply:
I'm always uncertain if a registering decorator a good idea. It makes the registration more intransparent. On the other hand it can be more semantic. In the present case I tend to think it's an improvement.

What I don't like is the `_style_list` param. At best we could get rid of it. However, I don't think that's possible (there is no such thing a decorator owned by the class, that can access the class variable `_style_list`. At least, can we rename the `_style_list` class variables to `_styles` or even better `_style_classes`.

Reply:
Uh, oh. `style_list` isn't even a list. 👿